### PR TITLE
Integrated pytest test suite into pyproject.toml

### PR DESCRIPTION
**Pull request by MitchiLaser**
The pytest test suite has been integrated into the pyproject.toml file. The following changes were made:

- The pytest.ini file was completely integrated into pyproject.toml
- The .coveragery file was completely integrated into the pyroject.toml
- the output files of a pytest run were added to the .gitignore
- a target to execute the tests and run coverage was added to the Makefile
- The github workflow to run the pytest test suite was restructured.

As the following [workflow run](https://github.com/MitchiLaser/kafe2/actions/runs/5831975152) shows: The pytest workflows run without causing issues.

Furthermore I took the opportunity to add myself to the list of maintainers for this project.

### CODE REVIEW

File: .gitignore
Comment (cverstege) [inline .gitignore]: Those are kafe2 files created during the tests to check the write to and load from file features including kafe2go.

File: pyproject.toml
Comment (cverstege) [inline pyproject.toml:13]: We have to define what our policy here will be. You are definitely shown as a contributor on the GitHub page, since you have commits in the code.
The authors of the python package might be more prone to receiving theoretical question about fitting in general. Not sure if we want to include every contributor here or keep this limited. This needs to be discussed.

Comment (MitchiLaser) [reply pyproject.toml:13]: OK, I can totally understand this point. Should I remove this change from the file until you are sure how to deal with this?

File: .gitignore
Comment (MitchiLaser) [reply .gitignore]: I changed this comment, hopefully the description is now fine

**Comment by MitchiLaser** [comment]
Are there any further changes to be done or is this pull request fine and just waiting for an approval?

**Comment by cverstege** [comment]
Looks good to me :+1:

**Review by JohannesGaessler — APPROVED** [review]
Looks good to me.

**Pull request merged**
